mmeval/metrics/ms_ssim.py

```diff
@@ -0,0 +1,294 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+import numpy as np
+from scipy import signal
+from typing import Dict, List, Sequence, Tuple
+
+from mmeval.core import BaseMetric
+from .utils.image_transforms import reorder_image
+
+
+class MultiScaleStructureSimilarity(BaseMetric):
+    """MS-SSIM (Multi-Scale Structure Similarity) metric.
+
+    Ref:
+    This class implements Multi-Scale Structural Similarity (MS-SSIM) Image
+    Quality Assessment according to Zhou Wang's paper, "Multi-scale structural
+    similarity for image quality assessment" (2003).
+    Link: https://ece.uwaterloo.ca/~z70wang/publications/msssim.pdf
+
+    Author's MATLAB implementation:
+    http://www.cns.nyu.edu/~lcv/ssim/msssim.zip
+
+    PGGAN's implementation:
+    https://github.com/tkarras/progressive_growing_of_gans/blob/master/metrics/ms_ssim.py
+
+    Args:
+        input_order (str): Whether the input order is 'HWC' or 'CHW'.
+            Defaults to 'HWC'.
+        max_val (int): the dynamic range of the images (i.e., the difference
+            between the maximum the and minimum allowed values).
+            Defaults to 255.
+        filter_size (int): Size of blur kernel to use (will be reduced for
+            small images). Defaults to 11.
+        filter_sigma (float): Standard deviation for Gaussian blur kernel (will
+            be reduced for small images). Defaults to 1.5.
+        k1 (float): Constant used to maintain stability in the SSIM calculation
+            (0.01 in the original paper). Defaults to 0.01.
+        k2 (float): Constant used to maintain stability in the SSIM calculation
+            (0.03 in the original paper). Defaults to 0.03.
+        weights (List[float]): List of weights for each level. Defaults to
+            [0.0448, 0.2856, 0.3001, 0.2363, 0.1333]. Noted that the default
+            weights don't sum to 1.0 but do match the paper / matlab code.
+        **kwargs: Keyword parameters passed to :class:`BaseMetric`.
+
+    Examples:
+
+        >>> from mmeval import MultiScaleStructureSimilarity as MS_SSIM
+        >>> import numpy as np
+        >>>
+        >>> ms_ssim = MS_SSIM()
+        >>> preds = [np.random.randint(0, 255, size=(3, 32, 32)) for _ in range(4)]  # noqa
+        >>> ms_ssim(preds)  # doctest: +ELLIPSIS
```

Comment:
Should groundtruths are passed in `ms_ssim`?

Reply:
Yes, I forget to update the example in docstring.